components/MessagePreviewCard/index.js

```diff
@@ -0,0 +1,96 @@
+import styled from "styled-components";
+
+
+export default function MessagePreviewCard( id, mutate,
+  name,
+  text,
+ date
+ ) {
+  async function handleDelete() {
+    const responseMessage = await fetch(`/api/messages/${id}`, {
+      method: "DELETE",
+    });
+    if (!responseMessage.ok) {
+      return <h1>Something gone wrong!</h1>;
+    }
+
+    if (responseMessage.ok) {
+        mutate()
```

Comment:
Usually, you don't need `mutate` if you navigate away because the navigation itself triggers a revalidation.